src/main/java/org/onebusaway/alexa/AuthedSpeechlet.java

```diff
@@ -73,6 +74,7 @@ public SpeechletResponse onIntent(final IntentRequest request,
                             userData.getCity(), userData.getRegionName()));
             return SpeechletResponse.newTellResponse(out);
         } else if (SET_STOP_NUMBER.equals(intent.getName())) {
+            populateAttributes(session);
```

Comment:
Agreed - I just force-pushed a new commit into this PR that moves `populateAttributes()` to the top of the `onIntent()` method, so the session will always contain the latest persisted user data.